src/restore/test/table_restore_invoker.py

```diff
@@ -22,8 +30,17 @@ def invoke(self, src_table_reference, target_dataset,
         resp, content = self.http.request(url)  # pylint: disable=W0612
         logging.info("Response: %s", resp)
         resp_data = json.loads(content)
-        logging.info("Restored table details: %s", resp_data)
-        return resp_data
+        return resp_data['restorationJobId']
+
+    @retry(JobInProgressException, tries=6, delay=10, backoff=2)
+    def wait_till_done(self, restoration_job_id):
+        result = RestorationJobStatusService()\
+            .get_restoration_job(restoration_job_id)
+
+        if result["status"]["state"] in "In progress":
```

Comment:
Periodical retrieving state using exceptions looks really bad. Could you use simple while loop?

Reply:
Done.